charts/orchestrator/templates/sonataflow-uninstall-hook.yaml

```diff
@@ -0,0 +1,130 @@
+{{- if .Values.sonataFlowOperator.enabled }}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ .Release.Name }}-sonataflow-hook-cleanup
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded,hook-failed
+    "helm.sh/hook-weight": "-1"
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ .Release.Name }}-sonataflow-hook-cleanup
+  annotations:
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded,hook-failed
+    "helm.sh/hook-weight": "0"
+rules:
+  - apiGroups: 
+    - apiextensions.k8s.io
+    resources:
+    - customresourcedefinitions
+    verbs:
+    - get
+  - apiGroups:
+    - sonataflow.org
+    resources:
+    - sonataflowclusterplatforms
+    verbs:
+    - delete
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: {{ .Release.Name }}-sonataflow-hook-cleanup
+  annotations:
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded,hook-failed
+    "helm.sh/hook-weight": "0"
+subjects:
+  - kind: ServiceAccount
+    name: {{ .Release.Name }}-sonataflow-hook-cleanup
+    namespace: {{ .Release.Namespace }}
+roleRef:
+  kind: ClusterRole
+  name: {{ .Release.Name }}-sonataflow-hook-cleanup
+  apiGroup: rbac.authorization.k8s.io    
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ .Release.Name }}-sonataflow-hook-cleanup
+  namespace: {{ .Values.orchestrator.namespace }}
+  annotations:
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded,hook-failed
+    "helm.sh/hook-weight": "0"    
+rules:
+  - apiGroups:
+    - sonataflow.org
+    resources:
+    - sonataflowplatforms
```

Comment:
should we add the `sonataflowbuild` as well? the recommendation of the Orchestrator is to use a pre-built image. but what if the customer decides to use the `dev` profile?

Reply:
Not sure: As long as there are no `sonataflowbuild` CRs stored in the chart, the script will not interact with these objects. Maybe I'm missing something?

Reply:
no, we don't include this in the charts.